### PR TITLE
Added file hashing

### PR DESCRIPTION
**Pull request by rushitote**
## Description

#537

### CODE REVIEW

File: src/services/upload/livePhotoService.ts [hunk 1]
```diff
@@ -47,11 +47,17 @@ export function getLivePhotoFileType(
...
+    imageMetadata: Metadata,
+    videoMetadata: Metadata
+) {
     return {
         ...imageMetadata,
         title: getLivePhotoName(imageMetadata.title),
         fileType: FILE_TYPE.LIVE_PHOTO,
+        imageHash: imageMetadata.hash,
+        videoHash: videoMetadata.hash,
+        hash: undefined,
```
Comment (abhinavkgrd) [inline src/services/upload/livePhotoService.ts:60]: don't need this

Comment (abhinavkgrd) [reply src/services/upload/livePhotoService.ts:60]: we should make hash as optional

File: src/utils/upload/index.ts [hunk 2]
```diff
@@ -93,3 +94,10 @@ export function areFileWithCollectionsSame(
 ): boolean {
     return firstFile.localID === secondFile.localID;
 }
+
+export async function getFileHash(file: File | ElectronFile) {
+    const hash = await cryptoGenericHash(
+        new Uint8Array(await file.arrayBuffer())
```
Comment (abhinavkgrd) [inline src/utils/upload/index.ts:100]: let's keep this in crypto util

File: src/services/upload/livePhotoService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rushitote) [reply src/services/upload/livePhotoService.ts:60]: No, since we are destructuring `imageMetadata` there will be a `hash` present.

**Review by abhinavkgrd — APPROVED** [review]

**Pull request merged**